com.intuit.graphql.parent/com.intuit.graphql.tests/src/com/intuit/graphql/tests/InBuiltDirectiveValidationTest.xtend

```diff
@@ -0,0 +1,68 @@
+package com.intuit.graphql.tests
+
+import com.google.inject.Inject
+import com.intuit.graphql.graphQL.Document
+import com.intuit.graphql.graphQL.GraphQLPackage
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.util.ParseHelper
+import org.eclipse.xtext.testing.validation.ValidationTestHelper
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.^extension.ExtendWith
+import org.eclipse.xtext.testing.extensions.InjectionExtension
+
+@InjectWith(GraphQLInjectorProvider)
+@ExtendWith(InjectionExtension)
+class InBuiltDirectiveValidationTest {
+
+	@Inject
+	ParseHelper<Document> parseHelper
+	
+	@Inject 
+	ValidationTestHelper validationTestHelper
+
+	@Test
+	def void deprecatedDirectiveDefined() throws Exception{
+		
+		val document = '''
+			type  Foo1 {
+				foo1: String  @deprecated(reason : "reason `in markdown syntax`")
+			}
+			
+			directive @deprecated(reason: String = "No longer supported") on FIELD_DEFINITION | ENUM_VALUE
+			
+		'''
+
+		parseHelper.parse(document)
+	}
+	
+	@Test
+	def void deprecatedDirectiveNotDefined() throws Exception{
+		
+		val document = '''
+			type  Foo1 {
+				foo1: String  @deprecated(reason : "reason `in markdown syntax`")
+			}
+		'''
+
+		parseHelper.parse(document)
+
+	}
+	
+	@Test
+	def void deprecatedDirectiveDefinedTwice() {
```

Comment:
@ashpak-shaikh  For some reason, few of these tests are failing in my local. Could you confirm if they are passing? 
`Failures: 
  InBuiltDirectiveValidationTest.deprecatedDirectiveDefinedTwice:78 expected:<[Duplicate directives in schema: deprecated]> but was:<[Expected ERROR 'null' on DirectiveDefinition at [-1:-1] but got
]>
  UnionTypeValidationTest.testUnionWithInterfaceMemberError:57 expected:<[Offending member type in union 'InterfaceFoo']> but was:<[Expected ERROR 'null' on UnionTypeDefinition at [-1:-1] but got
]>
  UnionTypeValidationTest.testUnionWithScalareMemberError:94 expected:<[Offending member type in union 'String']> but was:<[Expected ERROR 'null' on UnionTypeDefinition at [-1:-1] but got
]>`